modular_skyrat/modules/gunsgalore/code/ammo/ammo.dm

```diff
@@ -63,3 +71,175 @@
 	armour_penetration = 40
 	wound_falloff_tile = 0
+
+/obj/item/ammo_casing/realistic/a762x39/ricochet
+	name = "5.6x40mm match bullet casing"
+	desc = "A 5.6x40mm match bullet casing."
+	extended_desc = "MATCH: Ricochets everywhere. Like crazy."
+	projectile_type = /obj/projectile/bullet/a762x39/ricochet
+
+/obj/projectile/bullet/a762x39/ricochet
+	name = "5.6mm match bullet"
+	damage = 30
+	wound_bonus = 25
+	armour_penetration = 25
+	ricochets_max = 2
+	ricochet_chance = 100
+	ricochet_auto_aim_angle = 30
+	ricochet_auto_aim_range = 15
+	ricochet_incidence_leeway = 40
+	ricochet_decay_damage = 1
+	ricochet_shoots_firer = FALSE
+
+/obj/item/ammo_casing/realistic/a762x39/fire
+	name = "5.6x40mm incendiary bullet casing"
+	desc = "A 5.6x40mm incendiary bullet casing."
+	extended_desc = "TARGETED INCENDIARY: Leaves no trail when shot, sets targets aflame."
+	projectile_type = /obj/projectile/bullet/incendiary/a762x39
+
+/obj/projectile/bullet/incendiary/a762x39
+	name = "5.6mm incendiary bullet"
+	damage = 30
+	wound_bonus = 25
+	armour_penetration = 30
+	wound_falloff_tile = -5
+	fire_stacks = 2
+	leaves_fire_trail = FALSE
+
+/obj/item/ammo_casing/realistic/a762x39/xeno
+	name = "5.6x40mm anti-acid bullet casing"
+	desc = "A 5.6x40mm anti-acid bullet casing. Special chemical treatment and an additional layer of water-absorbent materials dissipates and absorbs water from the target's body, making any acid-blooded target melt from the inside."
+	extended_desc = "ANTI-ACID: Deals additional damage against advanced sentient Xenomorphs."
+	projectile_type = /obj/projectile/bullet/a762x39/hunting/xeno
+
+/obj/projectile/bullet/a762x39/xeno
+	name = "5.6mm anti-acid bullet"
+	damage = 25
+	wound_bonus = 15
+	armour_penetration = 0
+		/// Bonus force dealt against certain mobs
+	var/nemesis_bonus_force = 25
+		/// List of mobs we deal bonus damage to
+	var/list/nemesis_path = list(/mob/living/carbon/alien)
+
+/obj/projectile/bullet/a762x39/xeno/prehit_pierce(mob/living/target, mob/living/carbon/human/user)
+	if(istype(target, nemesis_path))
+		damage += nemesis_bonus_force
+	.=..()
+
+/obj/item/ammo_casing/realistic/a762x39/ap
+	name = "5.6x40mm armor-piercing bullet casing"
+	desc = "A 5.6x40mm armor-piercing bullet casing."
+	extended_desc = "ARMOR PIERCING: Increased armor piercing capabilities. What did you expect?"
+	projectile_type = /obj/projectile/bullet/a762x39
+
+/obj/projectile/bullet/a762x39/ap
+	name = "5.6mm armor-piercing bullet"
+	damage = 30
+	wound_bonus = 15
+	armour_penetration = 60
+	wound_falloff_tile = 0
+
+/obj/item/ammo_casing/realistic/a762x39/emp
+	name = "5.6x40mm ion bullet casing"
+	desc = "A 5.6x40mm ion bullet casing."
+	extended_desc = "EMP: Produces an Electro-Magnetic Pulse on impact, damaging electronics severely."
+	projectile_type = /obj/projectile/bullet/a762x39/emp
+
+/obj/projectile/bullet/a762x39/emp
+	name = "5.6mm ion bullet"
+	damage = 25
+	wound_bonus = 15
+	armour_penetration = 15
+	var/emp_radius = 0
```

Comment:
0 emp radius

Reply:
Intentional - it still works the way it's meant to, by creating EMP pulses on the target tile for the pinpoint... EMP creation, duh.